### ISSUE TITLE
Task 'compileJava' not found when I add a new NDK module in Hello-libs

### ISSUE THREAD

**Issue by wanzhang** [googlesamples/android-ndk#245]
Hello-libs compiles OK in my computer. But when I add a new lib module 'gadd' in Hello-libs, I got this build error:
//////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
 Executing tasks: [:app:generateDebugSources, :app:generateDebugAndroidTestSources, :app:prepareDebugUnitTestDependencies, :app:mockableAndroidJar, :app:compileDebugSources, :app:compileDebugAndroidTestSources, :app:compileDebugUnitTestSources, :gadd:compileJava]

Configuration on demand is an incubating feature.
Hello-libs2

FAILURE: Build failed with an exception.

* What went wrong:
Task 'compileJava' not found in project ':gadd'.

* Try:
Run gradle tasks to get a list of available tasks. Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output.

BUILD FAILED

Total time: 1.641 secs
////////////////////////////////////////////////////////////////////////////////////////////////////////////////
Here is my build.gradle file:

///////////////////////////////////////////////////////////////////////////////////////////////////////////////
apply plugin: 'com.android.model.native'

model {
    android {
        compileSdkVersion = 23
        buildToolsVersion = '23.0.2'

        defaultConfig {
            minSdkVersion.apiLevel = 13
            targetSdkVersion.apiLevel = 23
            versionCode = 1
            versionName = '1.0'
        }
        ndk {
            moduleName = 'gadd'
            abiFilters.addAll(['armeabi', 'armeabi-v7a', 'arm64-v8a', 'x86', 'x86_64', 'mips', 'mips64']) //this is default
            ldLibs.addAll(['android', 'log'])
        }
    }
}
task(distributeLib, type : Copy) {
    // trigger build library
    dependsOn assemble
    into '../distribution/gadd/'
    from('src/main/jni/gadd.h') {
        into 'include/'
    }
    from('build/outputs/native/release/lib') {
        into 'lib/'
    }
}
///////////////////////////////////////////////////////////////////////////////////////////////////////////////////

I have no define comipleJava in my build.gradle file. By the way, the project compiles no error when using gradle build command. Is it a bug of Android Studio?

**Comment by ggfan** [comment]
could be.  which version of the studio is yours?

tried to add another lib into hello-libs, it works fine.  just pushed it to branch three-libs:
 https://github.com/googlesamples/android-ndk/tree/three-libs
you could check it out: sample hello-libs [ I will delete it in one or 2 days ]

mine is studio 2.2 preview 4, ndk-R12

**Comment by wanzhang** [comment]
My android studio is 2.1.2

**Comment by ggfan** [comment]
android studio version might matter. could you try with hello-libs in that branch with 2.1.2 see if it works? theoretically it should not since this is gradle-experimental plugin.  Thanks

**Comment by wanzhang** [comment]
Thanks ggfan. I think the problem is caused by my ndk module creation procedure. There is no ndk module selection in Create New Module dialog, so I selected a Java Library as my new module template. I deleted all the files created automatically, and add build script and c sources. These may cause the javaCompile task in Android Studio. Now I make a copy of gmath folder as gadd, and modify all the necessary files, both Android Studio and gradle build command work well.
